joblib/_parallel_backends.py

```diff
@@ -265,6 +265,9 @@ def effective_n_jobs(self, n_jobs):
         This also checks if we are attempting to create a nested parallel
         loop.
         """
+        if mp is None:
```

Comment:
`mp is None` can happen for mostly two reasons (joblib/_multiprocessing_helpers.py):
* JOBLIB_MULTIPROCESSING environment variable set to 0 in which case you kind of know what you are doing so IMO a warning is superfluous.
* locking is not available on the system (e.g. Amazon AWS Lambda IIRC) and you already get a warning here.